tools/docker/healthcheck.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 HOST="localhost"
-PORT=6379
+PORT=$(netstat -tuln | grep -oE ':::[0-9]+' | grep -oE '[0-9]+')
```

Comment:
hello @manojks1999 .  are you sure it's the right filter?. Have you tried running a docker image and run this command?

Reply:
I guess this won't work on the default container image.
```console
dri@srv1> docker run -it --rm ghcr.io/dragonflydb/dragonfly sh -c "netstat -tuln | grep -oE ':::[0-9]+' | grep -oE '[0-9]+'"
sh: 1: netstat: not found
```

Reply:
Just confirmed, this fix is incorrect. @danriedl  it is possible to exec as root to the already existing container:

`docker exec -it -u 0 dfserver /usr/bin/bash` and then install net-tools and run the command.

Reply:
Hi @romange , made a change in the regular expression of the grep command, and I have made a test on the alpine docker image I have shared the logs here, and keep this on hold I will make a test on ubuntu image, and give an update.

```
{
        "Id": "457e9699eb13317d840957027dac99cc2392b53c272d2fd3eb7a63383d1beb52",
        "Created": "2024-02-28T19:50:15.507886048Z",
        "Path": "entrypoint.sh",
        "Args": [
            "--port",
            "6767"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 54519,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2024-02-28T19:50:18.506419807Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2024-02-29T01:21:51.124907738+05:30",
                        "End": "2024-02-29T01:21:52.352651502+05:30",
                        "ExitCode": 0,
                        "Output": "+PONG\r\n"
                    },
                    {
                        "Start": "2024-02-29T01:22:22.356281589+05:30",
                        "End": "2024-02-29T01:22:23.661224958+05:30",
                        "ExitCode": 0,
                        "Output": "+PONG\r\n"
                    },
                    {
                        "Start": "2024-02-29T01:22:53.662909485+05:30",
                        "End": "2024-02-29T01:22:54.954001121+05:30",
                        "ExitCode": 0,
                        "Output": "+PONG\r\n"
                    },
                    {
                        "Start": "2024-02-29T01:23:24.95819494+05:30",
                        "End": "2024-02-29T01:23:26.327866369+05:30",
                        "ExitCode": 0,
                        "Output": "+PONG\r\n"
                    },
                    {
                        "Start": "2024-02-29T01:23:56.329411658+05:30",
                        "End": "2024-02-29T01:23:57.615890188+05:30",
                        "ExitCode": 0,
                        "Output": "+PONG\r\n"
                    }
                ]
            }
        }
}
```

Reply:
Testing this is delayed as building the docker image is taking time. I will update here once with proper testing in the Ubuntu image.

Reply:
Hi @romange below is the inspect log of ubuntu docker file, it seems working fine,

```
{
        "Id": "614ad12239194bb2cee2f22063b15eb85e4aa75604eefb596f78d57fc4db038c",
        "Created": "2024-02-29T02:51:27.781317762Z",
        "Path": "entrypoint.sh",
        "Args": [
            "--port",
            "6869"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 24812,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2024-02-29T02:51:28.912175782Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "healthy",
                "FailingStreak": 0,
                "Log": [
                    {
                        "Start": "2024-02-29T08:21:58.920502738+05:30",
                        "End": "2024-02-29T08:22:00.061591505+05:30",
                        "ExitCode": 0,
                        "Output": "+PONG\r\n"
                    }
                ]
            }
        }
}
```

Reply:
I just now found out this fails when we provide more than one argumental port.

```
docker run ubuntu-testing --port 5676 --memcached_port 4323

netstat -tul
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 0.0.0.0:4323            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:5676            0.0.0.0:*               LISTEN     

```

do we want here to only monitor the port 5676 as provided and also to monitor the other ports that are exposed?

Reply:
`cat /proc/1/cmdline` will get you the command line with all the arguments. 
But I think that it may be simpler to limit healthcheck heuristic to find a custom port the way you do now, but also 
support an environment variable `HEALTHCHECK_PORT`, that in case it is provided to container via `-e`, the healthcheck uses its value for the port to connect. 
so in the example above one would need to pass: `docker run -e HEALTHCHECK_PORT=5676 ubuntu-testing --port 5676 --memcached_port 4323`

Reply:
Hi @romange ,

Updated the health check file accordingly, and I have tested it is working as expected. If this is approved, later we need to change the documentation with the option as passing environment variable for health check.

Reply:
Thank you for fixing this issue!